custom/icds/location_reassignment/download.py

```diff
@@ -1,10 +1,129 @@
 import io
+import itertools
+from collections import defaultdict
+from copy import deepcopy
+
+from memoized import memoized
+from openpyxl import Workbook
+
+from corehq.apps.es import UserES
+from corehq.apps.locations.models import SQLLocation
 
 
 class Download(object):
-    def __init__(self, domain, parent_location_id, leaf_location_type):
-        pass
+    def __init__(self, location_id):
```

Comment:
Why was domain removed? Generally I think it's good not to assume ids are globally unique.

Reply:
`location_id` are marked as unique is model/db, so should be safe here?

Reply:
They are right now, and I don't think they're likely to change, but I also wouldn't want to go through something like https://github.com/dimagi/commcare-hq/issues/25718 again.

Reply:
That would be some deep architectural change, and i hope we would check all usages before making them non unique. Also there are other places that filter just with location_id.

Reply:
I'd bank on ids being globally unique (since `site_code` is the "unique on the domain" thing).  Still, I do like having `domain` in there to make sure the user doesn't inadvertently (or maliciously) change something on another domain.  On ICDS in particular where there are test domains on the same environment, I could see someone getting mixed up and copying an ID from the production domain into an upload intended for the staging domain.

Reply:
Only location load happening here is via location_id and then fetching all locations related to it.
The location is chosen via UI where someone would see locations for domain only.
Happy to add a check for domain as well.

Reply:
https://github.com/dimagi/commcare-hq/pull/26951/commits/d6144a7041fe316a6ec0135eb1c11ac3e854f777